python/cudf/cudf/core/dataframe.py

```diff
@@ -1751,7 +1781,10 @@ def as_gpu_matrix(self, columns=None, order="F"):
                 shape=(nrow, ncol), dtype=dtype, order=order
             )
             for colidx, inpcol in enumerate(cols):
-                dense = inpcol.to_gpu_array(fillna="pandas")
+                if inpcol.dtype == dtype:
+                    dense = inpcol.to_gpu_array(fillna="pandas")
+                else:
+                    dense = inpcol.astype(dtype).to_gpu_array(fillna="pandas")
```

Comment:
If you're typecasting to the same dtype that should be a zero op, so you can remove the check above and just always have it do the `astype(dtype)`.

Reply:
removed.